### PR TITLE
IBX-1455: Redesign Content Tree for OSS

### PR DESCRIPTION
**Pull request by GrabowskiM**
https://issues.ibexa.co/browse/IBX-1455

### CODE REVIEW

File: src/bundle/Resources/views/themes/admin/content/location_view.html.twig [hunk 1]
```diff
@@ -16,15 +16,14 @@
 {% block body_class %}ez-content-view{% endblock %}
 
 {% block content_tree %}
-    <div class="ez-content-tree-container ez-content-tree-container--expanded"
+    {% set content_tree_width = app.request.cookies.get('ibexa-tb-ibexa-content-tree-container-width') %}
+    {% set content_tree_width_style = 'width: %spx;'|format(content_tree_width) %}
+
+    <div class="ez-content-tree-container"
```
Comment (lucasOsti) [inline src/bundle/Resources/views/themes/admin/content/location_view.html.twig:22]: Shouldn't `ibexa` prefix be here?

File: src/bundle/ui-dev/src/modules/content-tree/components/content-tree/content.tree.js [hunk 2]
```diff
@@ -84,6 +91,15 @@ export default class ContentTree extends Component {
         }));
     }
 
+    toggleCollapseTree() {
+        const width = this.isTreeCollapsed() ? 320 : 96;
```
Comment (lucasOsti) [inline src/bundle/ui-dev/src/modules/content-tree/components/content-tree/content.tree.js:95]: could you please put these values ​​in a variable whose name says what it is

**Review by OstafinL — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/scss/ui/modules/universal-discovery/_tree.scss [hunk 3]
```diff
@@ -4,20 +4,14 @@
     position: relative;
 
     .m-tree {
-        width: initial !important; // TODO: remove when Content Tree refactored
+        width: initial !important;
```
Comment (dew326) [inline src/bundle/Resources/public/scss/ui/modules/universal-discovery/_tree.scss:7]: I think the rule should be removed when content tree is refactored not only todo ;)

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/admin.content.tree.js [hunk 4]
```diff
@@ -94,5 +99,8 @@
     contentTreeContainer.addEventListener('transitioncancel', handleContainerTransitionStop, false);
     contentTreeContainer.addEventListener('transitionend', handleContainerTransitionStop, false);
 
-    window.addEventListener('resize', handleViewportChange, { capture: false, passive: true });
+    doc.body.addEventListener('ibexa-tb-rendered:ibexa-content-tree', removeContentTreeContainerWidth);
```
Comment (tischsoic) [inline src/bundle/Resources/public/js/scripts/admin.content.tree.js:102]: Maybe `ibexa-tb-rendered:ibexa-content-tree` is correct, but it feels strange :D

File: src/bundle/Resources/public/js/scripts/admin.content.tree.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+    const CLASS_CONTENT_TREE_EXPANDED = 'ibexa-content-tree-container--expanded';
+    const CLASS_CONTENT_TREE_ANIMATE = 'ibexa-content-tree-container--animate';
     const CLASS_BTN_CONTENT_TREE_EXPANDED = 'ibexa-btn--content-tree-expanded';
     const token = doc.querySelector('meta[name="CSRF-Token"]').content;
     const siteaccess = doc.querySelector('meta[name="SiteAccess"]').content;
-    const contentTreeContainer = doc.querySelector('.ez-content-tree-container');
-    const contentTreeWrapper = doc.querySelector('.ez-content-tree-container__wrapper');
+    const contentTreeContainer = doc.querySelector('.ibexa-content-tree-container');
+    const contentTreeWrapper = doc.querySelector('.ibexa-content-tree-container__wrapper');
     const btn = doc.querySelector('.ibexa-btn--toggle-content-tree');
```
Comment (tischsoic) [inline src/bundle/Resources/public/js/scripts/admin.content.tree.js:10]:
```suggestion
    const contentTreeToggleBtn = doc.querySelector('.ibexa-btn--toggle-content-tree');
```

File: src/bundle/Resources/views/themes/admin/content/location_view.html.twig [hunk 6]
```diff
@@ -16,16 +16,15 @@
 {% block body_class %}ez-content-view{% endblock %}
 
 {% block content_tree %}
-    <div class="ez-content-tree-container ez-content-tree-container--expanded"
+    {% set content_tree_width = app.request.cookies.get('ibexa-tb-ibexa-content-tree-container-width') %}
+    {% set content_tree_width_style = 'width: %spx;'|format(content_tree_width) %}
```
Comment (tischsoic) [inline src/bundle/Resources/views/themes/admin/content/location_view.html.twig:20]: We can also do:
```suggestion
    {% set content_tree_width_style = "width: #{content_tree_width}px;" %}
```

File: src/bundle/Resources/public/scss/ui/modules/content-tree/_header.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+        .ibexa-icon + .ibexa-icon {
+            margin-right: calculateRem(2px);
+        }
+    }
+
+    &__name {
+        display: flex;
+        align-items: center;
+        font-size: calculateRem(18px);
```
Comment (lucasOsti) [inline src/bundle/Resources/public/scss/ui/modules/content-tree/_header.scss:25]:
```suggestion
        font-size: $ibexa-text-font-size-large;
```

File: src/bundle/Resources/public/scss/ui/modules/content-tree/_list.item.scss [hunk 8]
```diff
@@ -123,7 +177,7 @@ $list-item-height: calculateRem(20px);
     }
 
     &--has-sub-items {
-        > #{$list-item}__label {
+        > #{$list-item}__row {
```
Comment (lucasOsti) [inline src/bundle/Resources/public/scss/ui/modules/content-tree/_list.item.scss:180]: do we really need a sass variable here?

**Review by OstafinL — CHANGES_REQUESTED** [review]

**Review by dew326 — APPROVED** [review]

**Pull request closed without merging**